ruby_event_store/lib/ruby_event_store/client.rb

```diff
@@ -87,7 +97,7 @@ def enrich_event_metadata(event)
       metadata[:timestamp] ||= clock.()
       metadata.merge!(metadata_proc.call || {}) if metadata_proc
 
-      event.class.new(event_id: event.event_id, metadata: metadata, data: event.data)
+      # event.class.new(event_id: event.event_id, metadata: metadata, data: event.data)
```

Comment:
what's up with this comment?

Reply:
Changing metatadata inside the instance vs new instance with new metadata. Keeping the old behavior was not possible during refactorings. A separate task would be to bring back the old behavior if still wanted, but test it properly. It was not.